src/rulesets/oas/__tests__/info-contact.ts

```diff
@@ -1,11 +1,10 @@
 import { Spectral } from '../../../spectral';
-import { commonOasRules } from '../index';
-
-const ruleset = { rules: commonOasRules() };
+import * as ruleset from '../ruleset.json';
 
 describe('info-contact', () => {
   const s = new Spectral();
   s.addRules({
+    // @ts-ignore
```

Comment:
So apparently the reason why Typescript is complaining is because of the `type` property, which is an enum, while the JSON file is kind of arbitrary.

There are two solutions for that; the first one is cheating a little bit on typescript:

```ts
  s.addRules({
    'info-contact': Object.assign(ruleset.rules['info-contact'], {
      enabled: true,
      type: RuleType[ruleset.rules['info-contact'].type]
    }),
  });
```

The second one is how we should start doing things with external input:

```ts
describe('info-contact', () => {
  const s = new Spectral();

  RTV.decode(ruleset.rules['info-contact'].type).map(p => {
    s.addRules({
      'info-contact': Object.assign(ruleset.rules['info-contact'], {
        enabled: true,
        type: RuleType[p]
      }),
    })
  }).mapLeft(e => expect(e).not.toBeDefined())
```

Let's go with the former for now, but let's think about the latter as well.

Reply:
Thanks! Some regexp wizardy made the refactoring easy, I will post an update in a sec.